services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -0,0 +1,484 @@
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.route.service.Impl;
+
+import com.futurewei.alcor.common.enumClass.RouteTableType;
+import com.futurewei.alcor.common.exception.DatabasePersistenceException;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.exception.ResourcePersistenceException;
+import com.futurewei.alcor.route.exception.*;
+import com.futurewei.alcor.route.service.NeutronRouterService;
+import com.futurewei.alcor.route.service.NeutronRouterToSubnetService;
+import com.futurewei.alcor.route.service.RouterDatabaseService;
+import com.futurewei.alcor.route.service.RouterExtraAttributeDatabaseService;
+import com.futurewei.alcor.web.entity.port.PortEntity;
+import com.futurewei.alcor.web.entity.route.*;
+import com.futurewei.alcor.web.entity.subnet.SubnetEntity;
+import com.futurewei.alcor.web.entity.subnet.SubnetWebJson;
+import com.futurewei.alcor.web.entity.subnet.SubnetsWebJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class NeutronRouterServiceImpl implements NeutronRouterService {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    private NeutronRouterToSubnetService routerToSubnetService;
+
+    @Autowired
+    private RouterDatabaseService routerDatabaseService;
+
+    @Autowired
+    private RouterExtraAttributeDatabaseService routerExtraAttributeDatabaseService;
+
+    @Override
+    public NeutronRouterWebRequestObject getNeutronRouter(String routerId) throws ResourceNotFoundException, ResourcePersistenceException, CanNotFindRouter {
+        NeutronRouterWebRequestObject neutronRouterWebRequestObject = new NeutronRouterWebRequestObject();
+
+        Router router = this.routerDatabaseService.getByRouterId(routerId);
+        if (router == null) {
+            throw new CanNotFindRouter();
+        }
+
+        RouterExtraAttribute routerExtraAttribute = this.routerExtraAttributeDatabaseService.getByRouterExtraAttributeId(router.getRouterExtraAttributeId());
+
+        BeanUtils.copyProperties(router, neutronRouterWebRequestObject);
+        RouteTable routeTable = router.getRouteTable();
+        neutronRouterWebRequestObject.setRouteTable(routeTable);
+        if (routerExtraAttribute != null) {
+            BeanUtils.copyProperties(routerExtraAttribute, neutronRouterWebRequestObject);
+            neutronRouterWebRequestObject.setId(routerId);
+        }
```

Comment:
Why you need to set routerID when routerExtraAttribute != null? 
The routerID should be set when the router has or has not extra attribute.

Reply:
The ID of NeutronRouterWebRequestObject should be router_id, but I used the BeanUtils.copyProperties method twice. The first time I used it, NeutronRouterWebRequestObject_id = router_id; But the second time I used it, route_extra_attribute_id would overwrite the previous value, so I had to change it set back